src/execution/sql/functions/string_functions.cpp

```diff
@@ -9,20 +9,27 @@
 
 namespace terrier::execution::sql {
 
-void StringFunctions::Concat(StringVal *result, exec::ExecutionContext *ctx, const StringVal &left,
-                             const StringVal &right) {
-  if (left.is_null_ || right.is_null_) {
+void StringFunctions::Concat(StringVal *result, exec::ExecutionContext *ctx, const int64_t num_inputs,
+                             const StringVal *inputs) {
+  if (num_inputs == 0) {
     *result = StringVal::Null();
     return;
   }
 
-  const std::size_t length = left.GetLength() + right.GetLength();
+  std::size_t length = 0;
+  for (int64_t i = 0; i < num_inputs; i++) {
+    length += inputs[i].GetLength();
```

Comment:
You need to only count (and copy) non-NULL inputs.

Reply:
Updated